Modules/Sources/JetpackStats/Extensions/Calendar+Navigation.swift

```diff
@@ -0,0 +1,111 @@
+import Foundation
+
+extension Calendar {
+    enum NavigationDirection {
+        case backward
+        case forward
+
+        var systemImage: String {
+            switch self {
+            case .backward: "chevron.backward"
+            case .forward: "chevron.forward"
+            }
+        }
+    }
+
+    /// Navigates to the next or previous period from the given date interval.
+    ///
+    /// This method navigates by the length of the period for the given component.
+    /// For example, if the interval represents 6 months and the component is `.month`,
+    /// it will navigate forward or backward by 6 months.
+    ///
+    /// - Parameters:
+    ///   - interval: The date interval to navigate from
+    ///   - direction: Whether to navigate forward (.forward) or backward (.backward)
+    ///   - component: The calendar component to use for navigation
+    /// - Returns: A new date interval representing the navigated period
+    func navigate(_ interval: DateInterval, direction: NavigationDirection, component: Calendar.Component) -> DateInterval {
+        let multiplier = direction == .forward ? 1 : -1
+
+        // Calculate the offset based on the interval length and component
+        let offset = calculateOffset(for: interval, component: component)
+
+        // Navigate by the calculated offset
+        guard let newStart = date(byAdding: component, value: offset * multiplier, to: interval.start),
+              let newEnd = date(byAdding: component, value: offset * multiplier, to: interval.end) else {
+            assertionFailure("Failed to navigate \(component) interval by \(offset)")
+            return interval
+        }
+
+        return DateInterval(start: newStart, end: newEnd)
+    }
+
+    /// Calculates the number of units of the given component in the interval
+    private func calculateOffset(for interval: DateInterval, component: Calendar.Component) -> Int {
+        let components = dateComponents([component], from: interval.start, to: interval.end)
+        return switch component {
+        case .day: components.day ?? 1
+        case .weekOfYear: components.weekOfYear ?? 1
+        case .month: components.month ?? 1
+        case .quarter: components.quarter ?? 1
+        case .year: components.year ?? 1
+        default: 1
+        }
+    }
+
+    /// Determines if navigation is allowed in the specified direction
+    func canNavigate(_ interval: DateInterval, direction: NavigationDirection, minYear: Int = 2000, now: Date = .now) -> Bool {
```

Comment:
Can this function be implemented like this?

```swift
let minYearsAgo = ...
let validInterval = DateInteravl(start: minYearsAgo, end: now)
let dest = navigate(interval, direction)
return validInterval.intersects(dest)
```

Reply:
I'd probably keep the current implementation for now. It seems a bit more direct.